### ISSUE TITLE
Cannot unlock ECU - 8R2907115C 0002, box not found in .csv

### ISSUE THREAD

**Issue by kpematopa** [bri3d/VW_Flash#120]
Hello,
I am trying to unlock ECU 8R2907115C 0002 but I got errors that the ECU is not known. Can we add it in the supported boxes?

**Comment by kpematopa** [comment]
![Uploading IMG_5673.jpeg…]()

**Comment by kpematopa** [comment]
![IMG_5674](https://github.com/bri3d/VW_Flash/assets/21357272/09e42d1a-25c7-40db-b3e4-ea2ecd25c27f)
![IMG_5672](https://github.com/bri3d/VW_Flash/assets/21357272/f45cb759-5944-41bc-b113-5a3742164ecc)

**Comment by kpematopa** [comment]
I unlocked the ECU, but my part number is not listed in [box_codes.csv](https://github.com/bri3d/VW_Flash/blob/master/data/box_codes.csv)
I guess I cannot use CAL file, but the full unlocked flash file .bin
Unlocking of ECU was 15 mins with FL_8V0906259H__0001.frf

**Comment by bri3d** [comment]
Yes, the full file should work. I can add this file to box_codes.csv for next release; I was not aware of 8R2 (Q5 without FlexRay) box codes, only the Simos18.4 ones for which I never finished an unlocking patch.

**Comment by kpematopa** [comment]
Mine is 18.1, model year 2015/2016, it is the first Q5 but facelift. Very rare engine with 8 injectors. When I flash full bin file do I need to calculate checksums?

**Comment by bri3d** [comment]
Checksums should be calculated automatically by VW_Flash.

**Comment by aarons3** [comment]
Very cool, I never knew that existed either.

Checksums will be corrected automatically (all blocks) by VW_Flash.

**Comment by kpematopa** [comment]
Do you plan to add unlock patch of simos 18.41?

**Comment by kpematopa** [comment]
Is it possible to avoid flash counter to increase on every flash or it is a function of the MCU?

**Comment by bri3d** [comment]
re: 18.41, probably not, I have no car to test with and no flexray hardware to use for bench development. The port flashing patch should be quite simple (it's the same as the 18.10 one with some addresses changed) but testing it is very high risk. 

re: flash counter, it has nothing to do with the MCU, it's just a variable stored into a dflash channel. No personal plans to patch it though, warranty fraud isn't really my thing.

**Comment by FirewinX** [comment]
8S0906259N is also not in the box_codes.csv is it possible to unlock with FL_5G0906259Q__0005.frf?

**Comment by aarons3** [comment]
Yes, same exact process unlocking with the 5G090629Q_0005 frf
